pyscitt/pyscitt/key_vault_sign_client.py

```diff
@@ -120,7 +114,15 @@ def http_sign(self, data: bytes):
         pub_key = cert.public_key()
         assert isinstance(pub_key, (EllipticCurvePublicKey))
         key_size = pub_key.curve.key_size
-        signature_algorithm, hash_algorithm = ALGORITHMS[key_size]
+
+        if key_size == 256:
+            signature_algorithm, hash_algorithm = ("ES256", "sha256")
+        elif key_size == 384:
+            signature_algorithm, hash_algorithm = ("ES384", "sha384")
+        elif key_size == 521:
+            signature_algorithm, hash_algorithm = ("ES512", "sha384")
```

Comment:
Shouldn't this be sha512?

```suggestion
            signature_algorithm, hash_algorithm = ("ES512", "sha512")
```

Reply:
I tried leaving everything the same as it was without thinking about the values too much but I think your suggestion should be correct.

- [FIPS 186 section 6.1.1](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-5.pdf) says `It is recommended that the security strength associated with the bit length of n and the security 
strength of the hash function be the same unless an agreement has been made between 
participating entities to use a stronger hash function.` P-521 then should use the strength greater than 256 because bit length of the key is ≥ 512
- [NIST says that P521 is allowed to be used ECDSA see section 3.1](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-186.pdf) and mentions that the hash functions which meet strength ≥ 256 are SHA-512 and SHA3-512 in Table 3.
- [AKV source](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_enums.py#L48C14-L48C19) mentions that P521 is linked to sha512 hash function and references RFC7518, in another place [ES512 implementation is using SHA512](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_internal/algorithms/ecdsa.py#L65)
- [RFC7518](https://datatracker.ietf.org/doc/html/rfc7518) is for JSON web algorithms and also says that `ES512 | ECDSA using P-521 and SHA-512`